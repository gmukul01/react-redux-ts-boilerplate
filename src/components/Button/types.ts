import { HTMLProps, WithThemeProp } from '@utils/types';

export interface Props extends HTMLProps<HTMLButtonElement>, WithThemeProp {
    solid?: boolean;
    flat?: boolean;
    outlined?: boolean;
}
