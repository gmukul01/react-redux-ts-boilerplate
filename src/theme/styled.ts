import * as styledComponents from 'styled-components';
import { Theme } from './types';

// tslint:disable-next-line: no-duplicate-imports
import { ThemedStyledComponentsModule } from 'styled-components';
const { default: styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme } = styledComponents as ThemedStyledComponentsModule<
    Theme
>;

export { styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme };
