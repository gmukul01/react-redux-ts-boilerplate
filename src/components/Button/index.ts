import { ButtonStyled as Button } from './Button.styled';

export default Button;
