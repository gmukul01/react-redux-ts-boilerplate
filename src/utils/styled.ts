import * as styledComponents from 'styled-components';
import { Theme } from '../theme';

// tslint:disable-next-line: no-duplicate-imports
import { ThemedStyledComponentsModule } from 'styled-components';
const { default: styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme } = styledComponents as ThemedStyledComponentsModule<
    Theme
>;

export { styled, css, createGlobalStyle, keyframes, ThemeProvider, withTheme };
