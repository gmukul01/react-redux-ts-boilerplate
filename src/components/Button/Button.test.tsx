import 'jest-styled-components';
import React from 'react';
import { create } from 'react-test-renderer';

import { ButtonStyled as Button } from './Button.styled';

describe('<Button />', () => {
    it('should render properly with flat prop', () => {
        const button = create(<Button flat>Flat Button</Button>);
        expect(button).toMatchSnapshot();
    });
    it('should render properly with outlined prop', () => {
        const button = create(<Button outlined>Outlined Button</Button>);
        expect(button).toMatchSnapshot();
    });
    it('should render properly with solid prop', () => {
        const button = create(<Button solid>Solid Button</Button>);
        expect(button).toMatchSnapshot();
    });
});
