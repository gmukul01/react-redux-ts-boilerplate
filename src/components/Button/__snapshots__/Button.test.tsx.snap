// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly with flat prop 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  padding: 10px;
  margin: 8px;
  border-radius: 3px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #5392f9;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
>
  Flat Button
</button>
`;

exports[`<Button /> should render properly with outlined prop 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  padding: 10px;
  margin: 8px;
  border-radius: 3px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid #5392f9;
  color: #5392f9;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
>
  Outlined Button
</button>
`;

exports[`<Button /> should render properly with solid prop 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  padding: 10px;
  margin: 8px;
  border-radius: 3px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #5392f9;
  color: #ffffff;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
>
  Solid Button
</button>
`;
